src/shared/machine-id-setup.c

```diff
@@ -209,37 +244,49 @@ int machine_id_setup(const char *root, sd_id128_t machine_id, MachineIdSetupFlag
                 }
         }
 
-        fd = safe_close(fd);
+        /* Hmm, we couldn't or shouldn't write the machine-id to /etc/? So let's write it to /run/machine-id
+         * as a replacement */
 
-        /* Hmm, we couldn't or shouldn't write the machine-id to /etc?
-         * So let's write it to /run/machine-id as a replacement */
+        if (write_run_machine_id) {
+                _cleanup_free_ char *run = NULL;
 
-        run_machine_id = prefix_roota(root, "/run/machine-id");
+                r = chase("/run/", root, CHASE_PREFIX_ROOT, &run, &run_fd);
```

Comment:
CHASE_MKDIR_0755 ?

BTW, is it useful to write $root/run/machine-id when $root is non-empty ??

Reply:
> BTW, is it useful to write $root/run/machine-id when $root is non-empty ??

not too much, no. I kept this in place because the test case relies on this, but you are right, we should consider dropping that, and rework the test case, but i was too lazy for that for now, wanted to just keep this a 1:1 port for now.